actors/states/check.go

```diff
@@ -213,3 +214,40 @@ func CheckMinersAgainstPower(acc *builtin.MessageAccumulator, minerSummaries map
 		acc.Require(provingPeriodCron != nil, "miner %v has no proving period cron", addr)
 	}
 }
+
+func CheckDealStatesAgainstSectors(acc *builtin.MessageAccumulator, minerSummaries map[addr.Address]*miner.StateSummary, marketSummary *market.StateSummary) {
+	for dealID, deal := range marketSummary.Deals { // nolint:nomaprange
+		if deal.SectorStartEpoch == abi.ChainEpoch(-1) {
+			// deal hasn't been activated yet, make no assertions about sector state
+			continue
+		}
+
+		minerSummary, found := minerSummaries[deal.Provider]
+		if !found {
+			acc.Addf("provider %v for deal %d not found among miners", deal.Provider, dealID)
+			continue
+		}
+
+		sectorDeal, found := minerSummary.Deals[dealID]
+		if !found {
+			acc.Addf("deal %d not referenced in active sectors of miner %v", dealID, deal.Provider)
+			continue
```

Comment:
If a sector terminates, the sector might be gone but the deal still remain for deferred clean-up. But in that case I think we can assert the deal's SlashEpoch is set.

Reply:
good call